src/server.erl

```diff
@@ -30,40 +30,76 @@ database_connect() ->
 %% TODO: We shall remove the cookie given that this is a public game, lmao
 start(_, _) ->
     Connection = database_connect(),
-    Pid = spawn(?MODULE, handle_message, [Connection]),
+    Pid = spawn(?MODULE, handle_master, [Connection]),
     erlang:register(lyceum_server, Pid),
     {ok, Pid}.
 
-handle_message(Connection) ->
+stream_user_data(#{user_pid := UserPid, connection := Connection} = State) ->
+    {ok, Data} = epgsql:set_notice_receiver(Connection, self()),    
+    io:format("Data: ~p\n", [Data]),
+    %% UserPid ! {ok, Data},
+    stream_user_data(State).
+
+%% timeout_user(State) ->
+
+% TODO: We should not just keep passing Connections here, this is dangerous for interactive stuff
+handle_user(#{user_pid := UserPid, connection := Connection} = State) ->
+    receive	    
+	{list_characters, #{username := Username, email := Email}} ->
+	    io:format("Querying user's characters..."),
+	    Characters = character:player_characters(Username, Email, Connection),
+	    UserPid ! {ok, Characters};
+	{create_character, Character_Map} ->
+	    io:format("This character logged"),
+	    character:create(Character_Map, Connection),
+	    UserPid ! ok;
+	joining_world -> 
+	    case maps:is_key(State, stream_pid) of
+		true ->
+		    io:format("We tried to join two worlds at once, bruh. Stopping right now\n"),
+		    erlang:exit(double_joining);
+		false -> NewState = State#{handler_pid := self()}, 
+			 StreamPid = spawn(?MODULE, stream_user_data, [NewState]),
+			 NewState = State#{stream_pid := StreamPid},
+			 %% TimerOutPid = spawn(?MODULE, timeout_user, [NewState]),
+			 handle_user(NewState)
+	    end;
+	{update_character, Character_Map} ->
+	    io:format("Character will be updated"),
+	    character:updateTemp(Character_Map, Connection),
+	    UserPid ! ok;
+	Something ->
```

Comment:
The catch all clause will prevent the `retrieve_character` clause from ever being reached.